### PR TITLE
core, eth, trie, light: clean up trie interface

### PR DESCRIPTION
**Pull request by rjl493456442**
This PR cleans up/improves some trie interfaces:

- `TryGetAccount`, `TryUpdateAccount`, `TryDeleteAccount` accepts `common.Address` as the parameters, which is more intuitive to callers

### CODE REVIEW

File: core/state/trie_prefetcher.go [hunk 1]
```diff
@@ -336,11 +336,7 @@ func (sf *subfetcher) loop() {
 					if _, ok := sf.seen[string(task)]; ok {
 						sf.dups++
 					} else {
-						if len(task) == len(common.Address{}) {
-							sf.trie.TryGetAccount(task)
-						} else {
-							sf.trie.TryGet(task)
-						}
+						sf.trie.TryGet(task)
```
Comment (holiman) [inline core/state/trie_prefetcher.go:339]: Wait, how can this change be correct?

Comment (rjl493456442) [reply core/state/trie_prefetcher.go:339]: It's only for prefetching. 

`TryGetAccount` will load the node and decode it as an account
`TryGet` will only load the node

Actually they are equivalent.

**Comment by rjl493456442** [comment]
Let's hold-on this PR for a while.

I am not sure the change for `TryGetNode` is really needed.

File: eth/protocols/snap/handler.go [hunk 2]
```diff
@@ -523,7 +523,7 @@ func ServiceGetTrieNodesQuery(chain *core.BlockChain, req *GetTrieNodesPacket, s
 			if snap == nil {
 				// We don't have the requested state snapshotted yet (or it is stale),
 				// but can look up the account via the trie instead.
-				account, err := accTrie.TryGetAccountWithPreHashedKey(pathset[0])
+				account, err := accTrie.TryGetAccountWithHash(common.BytesToHash(pathset[0]))
```
Comment (holiman) [inline eth/protocols/snap/handler.go:526]:
```suggestion
				account, err := accTrie.TryGetAccountByHash(common.BytesToHash(pathset[0]))
```
?

File: eth/protocols/snap/handler.go [hunk 3]
```diff
@@ -417,7 +417,7 @@ func ServiceGetStorageRangesQuery(chain *core.BlockChain, req *GetStorageRangesP
 			if err != nil {
 				return nil, nil
 			}
-			acc, err := accTrie.TryGetAccountWithPreHashedKey(account[:])
+			acc, err := accTrie.TryGetAccountWithHash(account)
```
Comment (holiman) [inline eth/protocols/snap/handler.go:420]:
```suggestion
			acc, err := accTrie.TryGetAccountByHash(account)
```
?

**Review by holiman — APPROVED** [review]
LGTM, only minor suggestion to name them `ByHash` instead of `WithHash`, since it refers to whether we look it up by hash.

**Pull request merged**